Source/Scene/GroundPolygon.js

```diff
@@ -0,0 +1,424 @@
+
+        this._stencilPassCommand = undefined;
+        this._colorPassCommand = undefined;
+    };
+
+    var attributeLocations = {
+        positionHigh : 0,
+        positionLow : 1,
+        normal : 2
+    };
+
+    function getSurfaceDelta(ellipsoid, granularity) {
+        var refDistance = ellipsoid.maximumRadius;
+        return refDistance - (refDistance * Math.cos(granularity / 2.0));
+    }
+
+    var scratchPosition = new Cartesian3();
+    var scratchNormal = new Cartesian3();
+    var scratchDeltaNormal = new Cartesian3();
+
+    // TODO: More than one level of the polygon hierarchy
+    function createShadowVolume(polygon, context) {
+        var polygonHierarchy = polygon._polygonHierarchy;
+        var ellipsoid = polygon._ellipsoid;
+        var granularity = polygon._granularity;
+
+        var results = PolygonGeometryLibrary.polygonsFromHierarchy(polygonHierarchy);
+        var positions = results.polygons[0];
+        var hierarchy = results.hierarchy[0];
+
+        var bottomCap = PolygonGeometryLibrary.createGeometryFromPositions(ellipsoid, positions, granularity, false);
+        var bottomPositions = bottomCap.attributes.position.values;
+        var numBottomCapVertices = bottomPositions.length / 3;
+        var numCapVertices = numBottomCapVertices + numBottomCapVertices;
+        var bottomIndices = bottomCap.indices;
+        var numBottomIndices = bottomIndices.length;
+        var numCapIndices = numBottomIndices + numBottomIndices;
+
+        var outerRing = hierarchy.outerRing;
+        var tangentPlane = EllipsoidTangentPlane.fromPoints(outerRing, ellipsoid);
+        var positions2D = tangentPlane.projectPointsOntoPlane(outerRing);
+
+        var windingOrder = PolygonPipeline.computeWindingOrder2D(positions2D);
+        if (windingOrder === WindingOrder.CLOCKWISE) {
+            outerRing.reverse();
+        }
+
+        var wall = PolygonGeometryLibrary.computeWallGeometry(outerRing, ellipsoid, granularity, false);
+        var walls = [wall];
+        var numWallVertices = wall.attributes.position.values.length / 3;
+        var numWallIndices = wall.indices.length;
+
+        var holes = hierarchy.holes;
+        var i;
+        var j;
+
+        for (i = 0; i < holes.length; i++) {
+            var hole = holes[i];
+
+            tangentPlane = EllipsoidTangentPlane.fromPoints(hole, ellipsoid);
+            positions2D = tangentPlane.projectPointsOntoPlane(hole);
+
+            windingOrder = PolygonPipeline.computeWindingOrder2D(positions2D);
+            if (windingOrder === WindingOrder.COUNTER_CLOCKWISE) {
+                hole.reverse();
+            }
+
+            walls.push(PolygonGeometryLibrary.computeWallGeometry(hole, ellipsoid, granularity, false));
+            numWallVertices += wall.attributes.position.values.length / 3;
+            numWallIndices += wall.indices.length;
+            walls.push(wall);
+        }
+
+        var maxAlt = 8500.0; // TODO: get max alt of terrain
```

Comment:
See [here](https://github.com/AnalyticalGraphicsInc/cesium/issues/2172#issuecomment-74669808).  Note that we shouldn't need to use the bounding volume since `MinimumHeight` and `MaximumHeight` are part of [quantized-mesh](https://cesiumjs.org/data-and-assets/terrain/formats/quantized-mesh-1.0.html).